File: backend/src/main/java/com/ironlogic/modules/workout/dto/WorkoutSetInput.java
package com.ironlogic.modules.workout.dto;

import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

/**
 * Input DTO for one WorkoutSet during overwrite-save.
 *
 * @param setNo set number inside the exercise
 * @param weight optional recorded weight
 * @param reps optional recorded reps
 * @param durationSeconds optional duration-based value
 * @param restSeconds optional rest seconds
 * @param rpe optional RPE
 * @param rir optional RIR
 * @param isWarmup whether the set is warmup
 * @param isCompleted whether the set is completed
 */
public record WorkoutSetInput(
        /** setNo must be unique inside the same workout exercise. */
        @NotNull
        Integer setNo,

        /** Optional recorded weight. */
        BigDecimal weight,

        /** Optional recorded reps. */
        Integer reps,

        /** Optional duration for timed movements. */
        Integer durationSeconds,

        /** Optional rest duration. */
        Integer restSeconds,

        /** Optional rating of perceived exertion. */
        BigDecimal rpe,

        /** Optional reps in reserve. */
        Integer rir,

        /** Whether the set is warmup; null is treated as false by service. */
        Boolean isWarmup,

        /** Whether the set is completed; null is treated as true by service. */
        Boolean isCompleted
) {
}

File: frontend/lib/features/workout/domain/model/workout_set_input.dart
/// WorkoutSet 保存输入模型。
///
/// 该类属于 `workout/domain`，用于训练详情页把可编辑表单数据提交给仓储层。
class WorkoutSetInput {
  /// 创建组输入对象。
  const WorkoutSetInput({
    required this.setNo,
    this.weight,
    this.reps,
    this.durationSeconds,
    this.restSeconds,
    this.rpe,
    this.rir,
    this.isWarmup,
    this.isCompleted,
  });

  /// 组号。
  final int setNo;

  /// 重量。
  final double? weight;

  /// 次数。
  final int? reps;

  /// 持续秒数。
  final int? durationSeconds;

  /// 休息秒数。
  final int? restSeconds;

  /// RPE。
  final double? rpe;

  /// RIR。
  final int? rir;

  /// 是否热身组。
  final bool? isWarmup;

  /// 是否完成。
  final bool? isCompleted;
}
